homework/index.js

```diff
@@ -29,16 +32,62 @@
     });
     return elem;
   }
+  function app(x) {
+    x.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const select = createAndAppend('select', root, { id: 'select' });
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+    const div1 = createAndAppend('div', root, { class: 'tired' });
+    const list = createAndAppend('ul', div1, { id: 'ul' });
+    const rep = createAndAppend('li', list);
+    const des = createAndAppend('li', list);
+    const fork = createAndAppend('li', list);
+    const up = createAndAppend('li', list);
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+    const list2 = createAndAppend('ul', div1, { text: '' });
+    function renderContributer(cont1) {
+      for (let i = 0; i < cont1.length; i++) {
+        const link = createAndAppend('a', list2, {
```

Comment:
with this your HTML structure is as the following which makes it invalid HTML structure, the only allowed children for a `ul` are `li`, `ul` or `ol`.
reference: 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul
look for permitted content below the example.